include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp

```diff
@@ -35,51 +35,33 @@ struct distance_measure
         : measure(T())
     {}
 
-    bool is_small() const { return false; }
-    bool is_zero() const { return false; }
-    bool is_positive() const { return false; }
-    bool is_negative() const { return false; }
-};
-
-template <typename T>
-struct distance_measure_floating
-{
-    T measure;
-
-    distance_measure_floating()
-        : measure(T())
-    {}
+    T zero() const
+    {
+      static T zero = 0;
+      return zero;
+    }
 
     // Returns true if the distance measure is small.
     // This is an arbitrary boundary, to enable some behaviour
     // (for example include or exclude turns), which are checked later
     // with other conditions.
-    bool is_small() const { return std::abs(measure) < 1.0e-3; }
+    bool is_small() const { return geometry::math::abs(measure) < 1.0e-3; }
 
     // Returns true if the distance measure is absolutely zero
-    bool is_zero() const { return measure == 0.0; }
+    bool is_zero() const
+    {
+      return ! is_positive() && ! is_negative();
+    }
 
     // Returns true if the distance measure is positive. Distance measure
     // algorithm returns positive value if it is located on the left side.
-    bool is_positive() const { return measure > 0.0; }
+    bool is_positive() const { return measure > zero(); }
```

Comment:
Why not simply `T(0)` which would be shorter, more clear IMO and not require an additional function?

Reply:
It's updated.